builder/amazon/common/ssm_driver.go

```diff
@@ -210,39 +192,24 @@ func (d *SSMDriver) openTunnelForSession(ctx context.Context) error {
 	return nil
 }
 
-func isRetryableError(output string) bool {
-	retryableError := []string{
-		"connection refused",
-	}
-	for _, err := range retryableError {
-		if strings.Contains(output, err) {
-			return true
-		}
-	}
-	return false
-}
-
 // StopSession terminates an active Session Manager session
 func (d *SSMDriver) StopSession() error {
-	d.wg.Add(1)
-	defer d.wg.Done()
-
-	// Stop retry polling process to avoid unwanted retries at this point
-	d.closeRetryPolling <- true
-
 	if d.session == nil || d.session.SessionId == nil {
 		return fmt.Errorf("Unable to find a valid session to instance %q; skipping the termination step",
 			aws.StringValue(d.sessionParams.Target))
 	}
 
-	_, err := d.SvcClient.TerminateSession(&ssm.TerminateSessionInput{SessionId: d.session.SessionId})
+	// Store session id before we nullify any active sessions
+	sid := aws.StringValue(d.session.SessionId)
+
+	// Stop retry polling process to avoid unwanted retries at this point
+	d.session = nil
+	close(d.retryAfterTermination)
```

Comment:
I'd remove the set to avoid a race condition here:
```suggestion
	// Stop retry polling process to avoid unwanted retries at this point
	close(d.retryAfterTermination)
```